### PR TITLE
Fix CRD-Generator tests and examples

### PR DESCRIPTION
**Pull request by baloo42**
Fix CRD-Generator tests and examples for #2540

See also https://github.com/fabric8io/kubernetes-client/issues/6683

/cc @metacosm

### CODE REVIEW

File: operator-framework/src/test/java/io/javaoperatorsdk/operator/CRDMappingInTestExtensionIT.java [hunk 1]
```diff
@@ -43,7 +43,7 @@ void correctlyAppliesManuallySpecifiedCRD() {
   @Group("crd.example")
   @Version("v1")
   @Kind("Test")
-  private static class TestCR extends CustomResource<Void, Void> implements Namespaced {
+  public static class TestCR extends CustomResource<Void, Void> implements Namespaced {
```
Comment (metacosm) [inline operator-framework/src/test/java/io/javaoperatorsdk/operator/CRDMappingInTestExtensionIT.java:46]: Is this change necessary? I think we used different visibility on purpose to check if this was possible to do…

File: operator-framework/src/test/java/io/javaoperatorsdk/operator/config/runtime/TestCustomResource.java [hunk 2]
```diff
@@ -7,5 +7,5 @@
 
 @Group("sample.javaoperatorsdk")
 @Version("v1")
-class TestCustomResource extends CustomResource<Void, Void> implements Namespaced {
+public class TestCustomResource extends CustomResource<Void, Void> implements Namespaced {
```
Comment (metacosm) [inline operator-framework/src/test/java/io/javaoperatorsdk/operator/config/runtime/TestCustomResource.java:10]: Same as above

File: operator-framework/src/test/java/io/javaoperatorsdk/operator/CRDMappingInTestExtensionIT.java (hunk 1 above)
Comment (baloo42) [reply operator-framework/src/test/java/io/javaoperatorsdk/operator/CRDMappingInTestExtensionIT.java:46]: Found the root cause here https://github.com/fabric8io/kubernetes-client/blob/main/crd-generator/api-v2/src/main/java/io/fabric8/crdv2/generator/CustomResourceInfo.java#L168

`getDeclaredConstructor()` does only work with public classes.
